model/distro/distro.go

```diff
@@ -252,6 +265,29 @@ func ValidateBootstrapMethod(method string) error {
 	return nil
 }
 
+// ValidateCommunicationMethod checks that the communication mechanism is one of
+// the supported methods.
+func ValidateCommunicationMethod(method string) error {
+	if !util.StringSliceContains(ValidCommunicationMethods, method) {
+		return fmt.Errorf("'%s' is not a valid communication method", method)
+	}
+	return nil
+}
+
+// ValidateBootstrapAndCommunicationMethods checks that the bootstrap and
+// communication mechanisms are recognized, and that the combination of the two
+// is allowed.
+func ValidateBootstrapAndCommunicationMethods(bootstrap string, communication string) error {
+	catcher := grip.NewBasicCatcher()
+	catcher.Add(ValidateBootstrapMethod(bootstrap))
+	catcher.Add(ValidateCommunicationMethod(communication))
+	if bootstrap == BootstrapMethodLegacySSH && communication != CommunicationMethodLegacySSH ||
```

Comment:
I'd use a switch statement rather than the OR clause

Reply:
Done.